lib/api.js

```diff
@@ -68,6 +72,17 @@ function parseApiResponse(respData, callback) {
   callback(null, respData.result);
 }
 
+function pushOntoRetryQueue() {
+  retryQueue.push(arguments);
+
+  if (!retryHandle) {
+    retryHandle = setInterval(function() {
```

Comment:
The first time we get an ENOTFOUND error, we will push something on the queue and start up this periodic queue flushing. Forever more we will be running a timer that checks this queue at that frequency even if we have a good connection again. Do we want to instead cancel the timer and flush the queue in the event that makeApiRequest succeeds? I think having the timer running with basically a no-op when the queue is empty is not that big of a deal, but we should at least consider it.